app/cdap/components/hydrator/components/TopPanel/TopPanel.tsx

```diff
@@ -253,6 +261,14 @@ export const TopPanel = ({
   );
   return (
     <>
+      {showNoEditChangeError && (
+        <ErrorBanner
+          error="You have not made any changes to the pipeline, deployment is not allowed."
+          onClose={() => {
```

Comment:
I think there can be an edge case, when the user clicks edit - makes a change and then reverts the change. In this case we still deploy here right?

Reply:
Currently it will still block the deploy. Why do we want to allow it?

Reply:
No we want to block it. I am checking if that case is covered as well